bodyfetcher.py

```diff
@@ -158,7 +158,7 @@ def make_api_call_for_site(self, site):
                 # We don't want to go over the 100-post API cutoff, so take the last
                 # (100-len(new_post_ids)) from intermediate_posts
 
-                intermediate_posts = intermediate_posts[(100 - len(new_post_ids)):]
+                intermediate_posts = intermediate_posts[-(100 - len(new_post_ids)):]
```

Comment:
This is good, important, and better than what it is now, but will still result in us not scanning some posts under various conditions.

Currently, the the way it's intended is that if we need to catch up by more than `100 - len(new_post_ids)` posts, then we never scan the earlier posts (i.e. if the last thing we scanned was more than 100 post IDs ago, then we just ignore the posts which were earlier than 100 post IDs ago, unless they are in our `new_post_ids` list). Something along those lines is definitely needed, because we can easily have an SD instance come active after not scanning for days, weeks, or months. Having the instance try to catch up from the point where it last left off would be substantially too much.

OTOH, as currently intended, if more than 100 posts have happened on a site since any SD was last active and scanned that site, then we are going to miss posts. On a site like SO, that's not all that long (guesstimate as < 5 minutes during busy times) and is something which we probably experience on an irregular basis.

If we are not going to miss scanning any posts, then, in addition to other changes, this should be:
```
intermediate_posts = intermediate_posts[:(100 - len(new_post_ids))]
```
That will give us the *first* portion of posts which we missed, not skipping to the last portion.

A bit later, we will need:
```
self.previous_max_ids[site] = max(intermediate_posts)
```
instead of:
```
self.previous_max_ids[site] = max(new_post_ids)
```
That will result in us not missing any posts, but we will re-scan any posts which are in `new_post_ids` which have IDs > `max(intermediate_posts)`.

In addition:
* We will need to make sure that we re-queue the site to fetch more posts, because we still have more to get.
* We will need to also store the max postID which we've seen, so we know what the end target is.
* We will need to have a cut-off as to how many posts are too many to try to catch up and/or how much time from last scanning is too much.